### PR TITLE
🎤 Events: api, views and registration

### PR DESCRIPTION
**Pull request by arnemolland**

### CODE REVIEW

File: components/MailForm.tsx [hunk 1]
```diff
@@ -6,14 +6,14 @@ interface MailFormProps {
...
+  const [email, setEmail] = useState('')
 
   const subscribe = async (e: FormEvent) => {
     e.preventDefault()
 
-    const res = await fetch('/api/subscribe', {
+    const res = await fetch('/api/newsletter/subscribe', {
       body: JSON.stringify({
-        email: mail
+        email: email
```
Comment (sondregj) [inline components/MailForm.tsx]: {email}

File: components/RegistrationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const RegistrationForm: NextPage<RegistrationFormProps> = props => {
+  const [name, setName] = useState('')
+  const [company, setCompany] = useState('')
+  const [email, setEmail] = useState('')
+
+  const register = async (e: FormEvent) => {
+    e.preventDefault()
+
+    const res = await fetch(`/api/events/${props.id}/register`, {
+      body: JSON.stringify({
```
Comment (sondregj) [inline components/RegistrationForm.tsx]: {name, email, company}

File: pages/events/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { Layout, RegistrationForm } from '../../components'
+import { useRouter } from 'next/router'
+import css from '../../styles/main.scss'
+import { NextPage } from 'next'
+import { Event } from '../../typings'
+import 'isomorphic-unfetch'
+
+interface EventProps {
+  event: any
```
Comment (sondregj) [inline pages/events/[slug].tsx]: ☢️

Comment (arnemolland) [reply pages/events/[slug].tsx]: tru

File: components/RegistrationForm.tsx (hunk 2 above)
Comment (arnemolland) [reply components/RegistrationForm.tsx]: tru

File: pages/events.tsx [hunk 4]
```diff
@@ -1,20 +1,61 @@
...
+              </a>
+            ))}
           </div>
         </section>
       </Layout>
     </div>
   )
 }
+Events.getInitialProps = async function(context) {
+  const res = await fetch(`${process.env.host}/api/events`)
```
Comment (sondregj) [inline pages/events.tsx]: process.env.host == undefined?

Comment (arnemolland) [reply pages/events.tsx]: Burde være satt i .env etter readme, men legger inn en if

**Pull request merged**